events/models.py

```diff
@@ -795,10 +795,23 @@ class Meta:
         )
 
     def assign(self, event, child):
-        if self.child:
+        if (
+            self.child
+            or (  # To keep unique constraint for tuple (value, event)
+                TicketSystemPassword.objects.exclude(pk=self.pk)
+                .filter(value=self.value, event=event)
+                .exists()
+            )
+            or (  # To keep unique constraint for tuple (event, child)
+                TicketSystemPassword.objects.exclude(pk=self.pk)
+                .filter(event=event, child=child)
+                .exists()
+            )
+        ):
             raise PasswordAlreadyAssignedError("The password is already assigned.")
 
         self.assigned_at = timezone.now()
+        self.event = event
```

Comment:
I don't really understand why this function even has the parameter `event` 🙂 Unless I'm missing something, there shouldn't be any need to allow event changing here. A password should be tied to an event before it is assigned, and assigning a password should not change its event.

So I would just change the signature to `assign(self, child)`. Then the uniqueness checks should be also unnecessary.